File: problemas/iniciante/1847/1847.py
A, B, C = [int(x) for x in input().strip().split(' ')]

if (B < A and C >= B):
    print(":)")
elif (B > A and C <= B):
    print(":(")
elif (B > A and C > B and C - B < B - A):
    print(":(")
elif (B > A and C > B and C - B >= B - A):
    print(":)")
elif (B < A and C < B and B - C < A - B):
    print(":)")
elif (B < A and C < B and B - C >= A - B):
    print(":(")
elif (A == B):
    if (C > B):
        print(":)")
    else:
        print(":(")

File: problemas/iniciante/1847/1847.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let [A, B, C] = lines.shift().trim().split(' ').map((x) => parseInt(x));

if (B < A && C >= B) {
    console.log(":)");
} else if (B > A && C <= B) {
    console.log(":(");
} else if (B > A && C > B && C - B < B - A) {
    console.log(":(");
} else if (B > A && C > B && C - B >= B - A) {
    console.log(":)");
} else if (B < A && C < B && B - C < A - B) {
    console.log(":)");
} else if (B < A && C < B && B - C >= A - B) {
    console.log(":(");
} else if (A === B) {
    if (C > B) {
        console.log(":)");
    } else {
        console.log(":(");
    }
}